crates/uv-resolver/src/resolver/environment.rs

```diff
@@ -494,7 +494,7 @@ impl<'d> Forker<'d> {
         let mut envs = vec![];
         {
             let not_marker = self.marker.negate();
-            if !env_marker.is_disjoint(&not_marker) {
+            if !env_marker.is_disjoint(&not_marker) && !env_marker.is_conflicting() {
```

Comment:
It might be a pain... Right now, `env_marker.is_disjoint(&not_marker)` returns `true` if either of the expressions are false, but that doesn't include the `is_conflicting` definition here -- it only looks at the `FALSE` node.

Reply:
You could make `is_conflicting` into `simplify_conflicting(...) -> MarkerTree` that returns a FALSE node if it matched one of the impossible cases, that might simplify

Reply:
Do you think the approach makes sense though?

Reply:
we'd have to try it out, but it does sound more elegant in theory

Reply:
andrew's idea is even better than that